src/Backups/RestorerFromBackup.cpp

```diff
@@ -100,7 +100,7 @@ RestorerFromBackup::RestorerFromBackup(
     , context(context_)
     , process_list_element(context->getProcessListElement())
     , after_task_callback(after_task_callback_)
-    , create_table_timeout(context->getConfigRef().getUInt64("backups.create_table_timeout", 300000))
+    , create_table_timeout(context->getConfig()->getUInt64("backups.create_table_timeout", 300000))
```

Comment:
Why not the config in from `BackupsWorker`? This way if `backups.create_table_timeout` changes on a reload, the next restore will use the updated value while for other config values (the ones that used in `BackupsWorker`) this is not true. This might be confusing for users. I think in general the same behavior was happening before, so the current version is not worse than before, but it could improved.

If I misunderstood something, than sorry.

Reply:
Yes. But I'd like to keep the change simple for the moment :)